services/src/main/java/org/keycloak/authentication/requiredactions/UpdatePassword.java

```diff
@@ -29,27 +35,55 @@
 import org.keycloak.events.Errors;
 import org.keycloak.events.EventBuilder;
 import org.keycloak.events.EventType;
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakContext;
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.KeycloakSessionFactory;
 import org.keycloak.models.ModelException;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.RequiredActionConfigModel;
+import org.keycloak.models.RequiredActionProviderModel;
 import org.keycloak.models.UserCredentialModel;
 import org.keycloak.models.UserModel;
 import org.keycloak.models.utils.FormMessage;
 import org.keycloak.policy.MaxAuthAgePasswordPolicyProviderFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.provider.ProviderConfigurationBuilder;
 import org.keycloak.services.messages.Messages;
 import org.keycloak.services.validation.Validation;
 import org.keycloak.sessions.AuthenticationSessionModel;
+import org.keycloak.userprofile.ValidationException;
+import org.keycloak.validate.ValidationError;
 
-import jakarta.ws.rs.core.MultivaluedMap;
-import jakarta.ws.rs.core.Response;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 
 /**
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
  * @version $Revision: 1 $
  */
 public class UpdatePassword implements RequiredActionProvider, RequiredActionFactory {
+
     private static final Logger logger = Logger.getLogger(UpdatePassword.class);
+
+    private static final List<ProviderConfigProperty> CONFIG_PROPERTIES;
+
+    public static final String MAX_AUTH_AGE_KEY = "max_auth_age";
+
+    static {
+        List<ProviderConfigProperty> properties = ProviderConfigurationBuilder.create() //
+                .property() //
+                .name(MAX_AUTH_AGE_KEY) //
+                .label("Maximum Age of Authentication") //
+                .helpText("Configures the duration in seconds this action can be used after the last authentication before the user is required to re-authenticate.") //
```

Comment:
```suggestion
                .helpText("Configures the duration in seconds this action can be used after the last authentication before the user is required to re-authenticate. This parameter is used just in the context of AIA when the kc_action parameter is available in the request, which is for instance when user himself updates his password in the account console. When the 'Maximum Authentication Age' password policy is used in the realm, it's value has precedence over the value configured here.") //
```

Nitpick: just suggesting to add a bit more info to the helpText to clarify it is used just in the context of AIA and about the fact that password policy has preference.